### PR TITLE
Ability to declare a default "First broker login flow" per Realm

### LINKED ISSUE #25823: Ability to declare a default "First broker login flow" per Realm
### Description

As of today, an admin can declare default flows for the following flow types:
- browser flow
- registration flow
- direct grant flow
- reset credentials flow
- client authentication flow
- docker authentication flow

I want an admin to be able to declare a default flow for the following flow type:
- first broker login flow

For now the default `first broker login flow` is always the built-in `first broker login`. I want to make my custom `first broker login flow` the default one.

image

### Discussion

_No response_

### Motivation

I want any new registered `identity provider`'s to be bound to no `first broker login` (no override). When needed, I expect the realm to use the realm's default `first broker login`.

### Details

_No response_

### PR DESCRIPTION
**Pull request by reda-alaoui**
Fix #25823

### CODE REVIEW

File: js/apps/admin-ui/src/identity-providers/add/AdvancedSettings.tsx [hunk 1]
```diff
@@ -26,7 +26,8 @@ const LoginFlow = ({
   field,
   label,
   defaultValue,
-}: FieldProps & { defaultValue: string }) => {
+  labelForEmpty = "none",
```
Comment (jonkoops) [inline js/apps/admin-ui/src/identity-providers/add/AdvancedSettings.tsx:29]: Why would this need be customizable? Can't we get away with `None`?

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: js/apps/admin-ui/src/identity-providers/add/AdvancedSettings.tsx (hunk 1 above)
Comment (jonkoops) [reply js/apps/admin-ui/src/identity-providers/add/AdvancedSettings.tsx:29]: Ok thanks, no need to revert, I just wanted to understand the motivation.

File: js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties [hunk 2]
```diff
@@ -341,7 +341,7 @@ postBrokerLoginFlowAlias=Post login flow
 refreshTokenMaxReuse=Refresh Token Max Reuse
 partialExportHeaderText=Partial export allows you to export realm configuration, and other associated resources into a json file.
 clientScopes=Client scopes
-loadingRealms=Loading realms…
+loadingRealms=Loading realms\u2026
```
Comment (jonkoops) [inline js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:344]: If you need Unicode you can add it directly, we load these files as UTF-8.
```suggestion
loadingRealms=Loading realms…
```

File: js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties [hunk 3]
```diff
@@ -1488,7 +1490,7 @@ encryptionAlgorithm=Encryption Algorithm
 requiredForLabel.users=Only users
 groupUpdated=Group updated
 hideMetaData=Hide metadata
-customAttribute=Custom Attribute…
+customAttribute=Custom Attribute\u2026
```
Comment (jonkoops) [inline js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:1493]:
```suggestion
customAttribute=Custom Attribute…
```

**Review by jonkoops — CHANGES_REQUESTED** [review]

File: js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties (hunk 2 above)
Comment (reda-alaoui) [reply js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties:344]: Sorry, while IntelliJ loads the file using UTF-8, it performs those changes automatically. I don't know why ...

Comment (jonkoops) [reply js/apps/admin-ui/maven-resources/theme/keycloak.v2/admin/messages/messages_en.properties]: It's likely for historical reasons, property files were not UTF-8 back in the day.

**Review by jonkoops — DISMISSED** [review]
LGMT front-end wise, can't judge the rest.

**Comment by reda-alaoui** [comment]
Could someone take a look at this please? :)

**Comment by pedroigor** [comment]
@reda-alaoui Wouldn't be easier to set the flow as a realm attribute? I understand it is not how we are doing for others but perhaps much better than having to change the database every time you need to bind a realm to a flow type.

@keycloak/core-iam wdty?

**Comment by reda-alaoui** [comment]
@pedroigor , I can do that.

Waiting for @keycloak/core-iam opinion

**Comment by reda-alaoui** [comment]
@pedroigor should I wait @keycloak/core-iam opinion before moving the new setting to a realm attribute?

**Comment by sguilhen** [comment]
In general, for entities that have associated "attribute" tables, we would add new attributes there instead of adding it directly to the entity in order to avoid migrations (both DB and model). One exception to this is when the new attribute is searchable, in which case it is generally better to have it as a standalone attribute in the entity and add an associated index to the new column.

**Comment by reda-alaoui** [comment]
@pedroigor , @sguilhen , the new attribute is now stored as a realm attribute.

**Review by sguilhen — COMMENTED** [review]
@reda-alaoui In general it looks good to me - I left a couple of really minor things I would adjust.

@pedroigor can you give this another review?

**Review by sguilhen — APPROVED** [review]
Thanks @reda-alaoui , it looks good to me now. 

@pedroigor This still needs a maintainer review, can you take a look?
@jonkoops as you were involved in this review, would you like to take a final look?

**Comment by reda-alaoui** [comment]
Ping :pray:

**Review by jonkoops — APPROVED** [review]
Front-end changes LGTM. Sorry for the late reply, had to work on some other stuff.

**Review by ahus1 — APPROVED** [review]
Approving as of @sguilhen's and @jonkoops' review.

**Comment by ahus1** [comment]
I figured out that squash-and-merge was still possible. So it is now merged!

**Pull request merged**